### DISCUSSION TITLE
Sync PostGIS layers with QFieldCloud: Security Issues

### DISCUSSION THREAD

**Discussion by nischmi1** [opengisch/QField#2468] [Q&A]
According to the docu (https://dev.qfield.cloud/documentation/advanced_setup.html#postgis), the PostGIS DB needs to be accessible to the world in order to be able to sync PostGIS Layers with the cloud. Together with the need to store username/password in clear-text in the qgis-project, I imagine that this is a "security-no-go" for many organizations.

Is an alternative, safer approach planned for the future, e.g. transferring gpkgs to the desktop and letting QFieldSync handle the (local) syncing to the DB? If yes, when?

Thanks a lot and cheers!
Nico

**Comment by suricactus** [comment]
Hello, @nischmi1 . The security-no-go you have seen is indeed something we are aware of. There are some planned improvements on this front. The one that is coming the next few months is uploading the `.pgservice` file on the cloud, so the credentials are not stored in plain text in the project file.

What you can already do is making your PG database open only for the IP range that app.qfield.cloud uses. 

The proposed solution with QFieldSync syncing the PG database locally is not planned for now.

**Comment by marioba** [comment]
In order to provide a clearer and more manageable discussion forum and to have all discussions about QField and related products (i.e. QFieldCloud and QFieldSync) in a single place, we transferred all the discussions from the QFieldCloud to the [QField repository](https://github.com/opengisch/QField/discussions).
